maestro-xcuitest-driver/src/main/kotlin/xcuitest/XCTestDriverClient.kt

```diff
@@ -177,4 +135,42 @@ class XCTestDriverClient(
 
         return okHttpClient.newCall(requestBuilder.build()).execute()
     }
+
+    private fun OkHttpClient.Builder.addInterceptors(): OkHttpClient.Builder {
+        val chain = addInterceptor(Interceptor {
+            val request = it.request()
+            try {
+                it.proceed(request)
+            } catch (connectException: IOException) {
+                if (restoreConnection()) {
+                    it.proceed(request)
+                } else {
+                    throw XCTestDriverUnreachable("Failed to reach out XCUITest Server")
+                }
+            }
+        })
+
+        if (installNetworkInterceptor) {
+            chain.addNetworkInterceptor(Interceptor {
+                val request = it.request()
+                try {
+                    it.proceed(request)
+                } catch (connectException: IOException) {
+                    // Fake an Ok response when shutting down and receiving an error
+                    // to prevent a stack trace in the cli.
+                    if (isShuttingDown) {
+                        Response.Builder()
+                            .request(it.request())
+                            .protocol(Protocol.HTTP_1_1)
+                            .code(200)
+                            .build()
+                    } else {
+                        throw XCTestDriverUnreachable("Failed to reach out XCUITest Server")
+                    }
```

Comment:
I assume the stack trace issue is why this network interceptor exists in the first place. Is that correct @amanjeetsingh150 ?

Reply:
Thats correct

Reply:
In that case, could we just add this network interceptor to both studio and non-studio runs? Seems quite harmless to fake the 200 while shutting down in the non-studio runs.

Reply:
I didn't saw usecase for shutting downs in non studio runs that's why avoided there

Reply:
Alright, I've simplified by always installing the shutdown handler.